kafka/source/pkg/reconciler/kafkasource.go

```diff
@@ -206,6 +219,188 @@ func checkResourcesStatus(src *v1alpha1.KafkaSource) error {
+			r.triggerLagThreshold = convertMapKeyToInt32(annotations, kedaTriggerLagThresholdAnnodationKey, logger)
+
+		}
+	}
+}
+
+func (r *Reconciler) generateKedaScaledObjectUnstructured(ctx context.Context, ra *v1.Deployment, src *v1alpha1.KafkaSource) (*unstructured.Unstructured, error) {
+	logger := logging.FromContext(ctx).Desugar()
+	deploymentName := ra.GetName()
+	namespace := src.Namespace
+	name := src.Name
+	logger.Info("Unstructured ScaledObject name "+name, zap.Any("name", name))
+	srcName := src.GetName()
+	srcUID := src.GetUID()
+	srcResVersion := src.GetResourceVersion()
+	logger.Info("Got srcResVersion="+srcResVersion, zap.Any("srcResVersion", srcResVersion))
+	srcKind := src.GetGroupVersionKind().Kind
+	logger.Info("Got srcKind srcName srcUID", zap.Any("srcKind", srcKind), zap.Any("srcName", srcName), zap.Any("srcUID", srcUID))
+	srcBrokerList := src.Spec.BootstrapServers
+	srcConcumerGroup := src.Spec.ConsumerGroup
+	triggers := make([]map[string]interface{}, 0, 1)
+	topics := strings.Split(src.Spec.Topics, ",")
+	if len(topics) == 0 {
+		return nil, fmt.Errorf("Comma-separated list of topics can not be empty")
+	}
+	for _, topic := range topics {
+		triggerMetadata := map[string]interface{}{
+			"brokerList":    srcBrokerList,
+			"consumerGroup": srcConcumerGroup,
+			"topic":         topic,
+		}
+		if r.triggerLagThreshold != nil {
+			logger.Info("Got triggerLagThreshold", zap.Any("triggerLagThreshold", r.triggerLagThreshold))
+			triggerMetadata["lagThreshold"] = strconv.Itoa(int(*r.triggerLagThreshold))
+		}
+		trigger := map[string]interface{}{
+			"type":     "kafka",
+			"metadata": triggerMetadata,
+		}
+		triggers = append(triggers, trigger)
+	}
+	spec := map[string]interface{}{
+		"scaleTargetRef": map[string]interface{}{
+			"deploymentName": deploymentName,
+		},
+		"triggers": triggers,
+	}
+	if r.minReplicaCount != nil {
+		logger.Info("Got minReplicaCount", zap.Any("minReplicaCount", r.minReplicaCount))
+		spec["minReplicaCount"] = *r.minReplicaCount
+	}
+	if r.maxReplicaCount != nil {
+		logger.Info("Got maxReplicaCount", zap.Any("maxReplicaCount", r.maxReplicaCount))
+		spec["maxReplicaCount"] = *r.maxReplicaCount
+	}
+	if r.cooldownPeriod != nil {
+		logger.Info("Got cooldownPeriod", zap.Any("cooldownPeriod", r.cooldownPeriod))
+		spec["cooldownPeriod"] = *r.cooldownPeriod
+	}
+	if r.pollingInterval != nil {
+		logger.Info("Got pollingInterval", zap.Any("pollingInterval", r.minReplicaCount))
+		spec["pollingInterval"] = *r.pollingInterval
+	}
+	soUnstr := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "keda.k8s.io/v1alpha1",
+			"kind":       "ScaledObject",
+			"metadata": map[string]interface{}{
+				"creationTimestamp": nil,
+				"namespace":         namespace,
+				"name":              name,
+				"labels": map[string]interface{}{
+					"deploymentName": deploymentName,
+				},
+				"ownerReferences": []map[string]interface{}{{
```

Comment:
I hit a problem in UTs when using `[]map[string]interface{}`. Some problem unmarshalling stuff. 
You should use `[]interface{}{map[string]interface{}`